### PR TITLE
 Add devfile for dashboard

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
Provide `devfile` for dashboard development, and new docker image for dashboard development

### What issues does this PR fix or reference?
#12228

### CODE REVIEW

File: dashboard/.devfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    actions:
+      - type: exec
+        tool: dash-dev
+        command: cd /projects/che/dashboard && yarn test
+        workdir: /projects/che/dashboard
+  - name: dashboard_dev_server
+    actions:
+      - type: exec
+        tool: dash-dev
+        command: cd /projects/che/dashboard && node_modules/.bin/gulp serve --server=${CHE_API_EXTERNAL%????}
```
Comment (benoitf) [inline dashboard/.devfile:44]: %???? is the correct syntax ?

Comment (evidolob) [reply dashboard/.devfile:44]: Yes, its in POSIX spec, see https://unix.stackexchange.com/a/259042

File: dockerfiles/dashboard-dev/src/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# Copyright (c) 2018-2018 Red Hat, Inc.
```
Comment (benoitf) [inline dockerfiles/dashboard-dev/src/entrypoint.sh]: minor on the date

File: dockerfiles/dashboard-dev/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2015-2018 Red Hat, Inc.
```
Comment (benoitf) [inline dockerfiles/dashboard-dev/Dockerfile]: minor : date

**Review by benoitf — APPROVED** [review]

**Pull request merged**